client.go

```diff
@@ -310,20 +297,27 @@ func (c *Client) runForever() {
 
 	go func() {
 		for {
-			c.debugLog("client: connecting...")
+			c.logger.Debug("client: connecting...")
 
 			err := c.runOnce()
 			if err == nil {
-				c.debugLog("client: finished gracefully")
+				c.logger.Debug("client: finished gracefully")
 				break
 			}
 
 			if atomic.LoadInt32(&c.wantStop) == 1 {
-				c.debugLog("client: finished with error %s", err.Error())
+				c.logger.Error("client: finished with error",
```

Comment:
Now that we use structured logging, should the `client`/`server` prefix be an attribute instead, maybe something like `component`? That will also make it even easier to filter logs.

Can you store default attributes on the logger on `c.logger` and `s.logger` so we only have to add it during setup?

Reply:
That is a great idea! We can store that on the logger.